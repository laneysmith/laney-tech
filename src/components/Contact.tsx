import React from 'react';
import { StaticQuery, graphql } from 'gatsby';

import Section from './Shared/Section';
import { List, ListItem } from './Shared/List';
import { SiteMetadata } from '../types/siteMetadata';

interface ContactProps {
  data: {
    site: {
      siteMetadata: {
        social: SiteMetadata['social'];
      };
    };
  };
}

export const PureContact: React.FC<ContactProps> = ({ data }) => {
  const { github, linkedin, email } = data.site.siteMetadata.social;
  const GITHUB_URL = `https://github.com/${github}`;
  const LINKEDIN_URL = `https://linkedin.com/in/${linkedin}`;
  const MAIL_TO = `mailto:${email}`;

  return (
    <Section title="Contact me">
      <List>
        <ListItem $icon="ðŸ’¾">
          <a href={GITHUB_URL}>github.com/{github}</a>
        </ListItem>
        <ListItem $icon="ðŸ‘”">
          <a href={LINKEDIN_URL}>linkedin.com/in/{linkedin}</a>
        </ListItem>
        <ListItem $icon="ðŸ“¨">
          <a href={MAIL_TO}>{email}</a>
        </ListItem>
      </List>
    </Section>
  );
};

const Contact = (): React.ReactElement => (
  <StaticQuery
    query={graphql`
      query ContactQuery {
        site {
          siteMetadata {
            social {
              github
              linkedin
              email
            }
          }
        }
      }
    `}
    render={data => <PureContact data={data} />}
  />
);

export default Contact;
