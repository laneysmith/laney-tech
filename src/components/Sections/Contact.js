import React from 'react';

import SectionTemplate from './SectionTemplate';
import List from './List';
import ListItem from './ListItem';

const Contact = () => (
  <SectionTemplate title="Contact Me">
    <List>
      <ListItem icon="ðŸ’»">
        <a href="https://github.com/laneysmith">github</a>
      </ListItem>
      <ListItem icon="ðŸ‘”">
        <a href="https://linkedin.com/in/laneysmith">linkedin</a>
      </ListItem>
      <ListItem icon="ðŸ“¨">
        <a href="mailto:lsmith.unc@gmail.com">email</a>
      </ListItem>
    </List>
  </SectionTemplate>
);

export default Contact;
