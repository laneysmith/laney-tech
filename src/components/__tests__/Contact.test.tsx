import React from 'react';
import { render, screen } from '@testing-library/react';
import { PureContact as Contact } from '../Contact';

const data = {
  site: {
    siteMetadata: {
      social: {
        github: 'my-github-username',
        linkedin: 'my-linkedin-username',
        email: 'me@theinternet.com',
      },
    },
  },
};

describe('Contact', () => {
  it('should display contact links', () => {
    render(<Contact data={data} />);

    expect(screen.getByRole('link', { name: 'github.com/my-github-username' })).toHaveAttribute(
      'href',
      'https://github.com/my-github-username'
    );
    expect(
      screen.getByRole('link', { name: 'linkedin.com/in/my-linkedin-username' })
    ).toHaveAttribute('href', 'https://linkedin.com/in/my-linkedin-username');
    expect(screen.getByRole('link', { name: 'me@theinternet.com' })).toHaveAttribute(
      'href',
      'mailto:me@theinternet.com'
    );
  });

  it('should match snapshot', () => {
    const data = {
      site: {
        siteMetadata: {
          social: {
            github: 'my-github-username',
            linkedin: 'my-linkedin-username',
            email: 'me@theinternet.com',
          },
        },
      },
    };
    const { container } = render(<Contact data={data} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
