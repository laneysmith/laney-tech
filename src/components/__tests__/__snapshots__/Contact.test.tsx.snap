// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact should match snapshot 1`] = `
.c0 {
  margin: 0 0 0.8125rem 0;
  padding: 0 0.8125rem;
}

.c1 {
  padding: 0.8125rem 0 0.8125rem 0.8125rem;
  margin-left: 0.8125rem;
  border-left: 1px solid;
}

.c2 {
  list-style: none;
  padding: 0;
  margin: 0.40625rem 0 0.40625rem 0.40625rem;
}

.c3 {
  position: relative;
  margin-left: 1.625rem;
}

.c3::before {
  content: 'ðŸ’¾';
  position: absolute;
  left: -1.625rem;
}

.c4 {
  position: relative;
  margin-left: 1.625rem;
}

.c4::before {
  content: 'ðŸ‘”';
  position: absolute;
  left: -1.625rem;
}

.c5 {
  position: relative;
  margin-left: 1.625rem;
}

.c5::before {
  content: 'ðŸ“¨';
  position: absolute;
  left: -1.625rem;
}

@media only screen and (max-width:600px) {
  .c0 {
    padding: 0;
  }
}

<section
  class="c0"
>
  <h2>
    Contact me
  </h2>
  <div
    class="c1"
  >
    <ul
      class="c2"
    >
      <li
        class="c3"
      >
        <a
          href="https://github.com/my-github-username"
        >
          github.com/
          my-github-username
        </a>
      </li>
      <li
        class="c4"
      >
        <a
          href="https://linkedin.com/in/my-linkedin-username"
        >
          linkedin.com/in/
          my-linkedin-username
        </a>
      </li>
      <li
        class="c5"
      >
        <a
          href="mailto:me@theinternet.com"
        >
          me@theinternet.com
        </a>
      </li>
    </ul>
  </div>
</section>
`;
